tests/tcp.rs

```diff
@@ -1188,3 +1191,41 @@ fn write_then_deregister() {
     assert_eq!(s.read(&mut buf).unwrap(), 4);
     assert_eq!(&buf[0..4], &[1, 2, 3, 4]);
 }
+
+#[test]
+fn tcp_no_events_after_deregister() {
+    let (mut poll, mut events) = init_with_poll();
+
+    let listener = TcpListener::bind("127.0.0.1:0".parse().unwrap()).unwrap();
+    let addr = listener.local_addr().unwrap();
+    let mut stream = TcpStream::connect(addr).unwrap();
+
+    poll.registry()
+        .register(&listener, Token(1), Interests::READABLE)
+        .unwrap();
+    poll.registry()
+        .register(&stream, Token(3), Interests::READABLE)
+        .unwrap();
+
+    // Wait a moment for stream to connect before trying accept
+    thread::sleep(Duration::from_millis(100));
```

Comment:
Why aren't you polling here?

Reply:
Wanted to not exhaust the event by polling it, and keep it for later check after deregister, but I guess there are enough checks, will do polling.